internal/charmstore/store.go

```diff
@@ -1284,7 +1286,13 @@ func (s *Store) DeleteEntity(id *router.ResolvedURL) error {
 			return errgo.Notef(err, "cannot remove compatibility blob %s", name)
 		}
 	}
-
+	if s.ES == nil || s.ES.Database == nil {
```

Comment:
you don't need to do this check in both places, just call s.ES.delete and let it return nil if its not configured.